clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3203,7 +3204,14 @@ private int processKafkaDataMessage(
             ChunkedValueManifest chunkedValueManifest = manifestSerializer.deserialize(
                 ByteUtils.extractByteArray(put.getPutValue()), // must be done before recordTransformer changes putValue
                 AvroProtocolDefinition.CHUNKED_VALUE_MANIFEST.getCurrentProtocolVersion());
-            hostLevelIngestionStats.recordAssembledValueSize(chunkedValueManifest.getSize(), currentTimeMs);
+            ByteBuffer rmdPayload = put.getReplicationMetadataPayload();
+            boolean isValueManifest = rmdPayload != null && rmdPayload.equals(VeniceWriter.EMPTY_BYTE_BUFFER);
+            if (isValueManifest) {
+              recordAssembledRecordSizeRatio(keyLen + chunkedValueManifest.getSize(), currentTimeMs);
```

Comment:
I just noticed this statement:
```
recordAssembledRecordSizeRatio(keyLen + chunkedValueManifest.getSize(), currentTimeMs);
```
This seems to be incorrect, as we would like to emit the ratio, not the absolute size.
Shouldn't be:
```
recordAssembledRecordSizeRatio((keyLen + chunkedValueManifest.getSize() / maxRecordSizeBytes)  , currentTimeMs);
```

Or override the implementation in `LFSIT`?

Reply:
Yeah, since `maxRecordSizeBytes` needs `VeniceWriter`, the setting is only present on `LFSIT`. The implementation is overridden in `LFSIT`, but I do agree that it is confusing with the current naming:
```java
  @Override
  protected final void recordAssembledRecordSizeRatio(long size, long currentTimeMs) {
    if (maxRecordSizeBytes != VeniceWriter.UNLIMITED_MAX_RECORD_SIZE) {
      hostLevelIngestionStats.recordAssembledRecordSizeRatio((double) size / maxRecordSizeBytes, currentTimeMs);
    }
  }
```

Reply:
Sure, that will be better.